src/pages/editProfile/index.js

```diff
@@ -0,0 +1,229 @@
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import TextInput from "../../components/form/textInput";
+import Button from "../../components/button";
+import ProfileCircle from "../../components/profileCircle";
+import Card from "../../components/card";
+import { useNavigate } from "react-router-dom";
+import useModal from "../../hooks/useModal";
+import ErrorMessage from "../../components/errorMessage";
+import "./style.css";
+import SaveChangesModal from "../../components/saveChangesModal"
+import { get } from "../../service/apiClient";
+import { emptyProfile } from "../../service/mockData";
+import useAuth from "../../hooks/useAuth";
+import ProfileImg from "../../components/profileImage/ProfileImage";
+
+
+const EditProfile = () => {
+  const [profile, setProfile] = useState(emptyProfile);
+  const [formState, setFormState] = useState(profile);
+  const [readOnly, setReadOnly] = useState(false)
+  const [passwordPermission, setPasswordPermission] = useState(true)
+  const [isError, setIsError] = useState(false);
+  const navigate = useNavigate()
+  const { id } = useParams()
```

Comment:
add a userId with a setter state

on the useEffect check if the id is different than the userid and update the userId